stream-compositions/services/legal-entity-composition-service/src/main/java/com/backbase/stream/compositions/legalentity/core/mapper/GrandCentralPartyMapper.java

```diff
@@ -0,0 +1,124 @@
+            || properties.getReferenceJobRoleNames().isEmpty()) {
+            throw new RuntimeException("Parent external id or reference job role names is empty");
+        }
+
+        legalEntity.setName(getPartyName(party));
+        legalEntity.setRealmName(properties.getRealmName());
+        legalEntity.setParentExternalId(properties.getParentExternalId());
+        if (Data.PartyType.PERSON.equals(party.getPartyType())) {
+            legalEntity.setUsers(List.of(new JobProfileUser(mapUser(party))
+                .referenceJobRoleNames(properties.getReferenceJobRoleNames())));
+        } else if (party.getCustomFields() != null) {
+            // If a party is not a person, the custom fields should be applicable for the LE, not the User
+            legalEntity.setAdditions(party.getCustomFields().getAdditions());
+        }
+    }
+
+    @AfterMapping
+    protected void updateUser(Data party, @MappingTarget User user) {
+        user.setFullName(getPartyName(party));
+        user.setIdentityLinkStrategy(properties.getIdentityUserLinkStrategy());
+        var email = getPartyEmail(party, Email.Type.PERSONAL, true)
+            .map(this::map)
+            .orElseThrow(() -> new RuntimeException("Email is required"));
+        user.setEmailAddress(email);
+        var mobilePhone = getPartyPhone(party, PhoneAddress.Type.MOBILE)
+            .map(this::map)
+            .orElseThrow(() -> new RuntimeException("Mobile phone is required"));
+        user.setMobileNumber(mobilePhone);
+        if (party.getStatus() != null) {
+            user.setLocked(!"ACTIVE".equals(party.getStatus()));
+        }
+        if (Data.PartyType.PERSON.equals(party.getPartyType()) && party.getCustomFields() != null) {
+            user.setAdditions(party.getCustomFields().getAdditions());
+        }
+    }
+
+    private String getPartyName(Data party) {
+        if (Data.PartyType.ORGANISATION.equals(party.getPartyType()) && party.getOrganisationName() != null) {
+            return party.getOrganisationName();
+        }
+        return getPersonFullName(party).orElseThrow(() -> new RuntimeException("Data name is required"));
+    }
+
+    private Optional<Email> getPartyEmail(Data party, Email.Type type, boolean fallbackToFirst) {
+        if (party.getElectronicAddress() == null || party.getElectronicAddress().getEmails() == null) {
+            return Optional.empty();
+        }
+        var selectedEmail = party.getElectronicAddress().getEmails().stream()
+            .filter(e -> type.equals(e.getType()))
+            .findFirst();
+        return fallbackToFirst && selectedEmail.isEmpty()
+            ? party.getElectronicAddress().getEmails().stream().findFirst()
+            : selectedEmail;
+    }
+
+    private Optional<PhoneAddress> getPartyPhone(Data party, PhoneAddress.Type type) {
+        if (party.getPhoneAddresses() == null) {
+            return Optional.empty();
+        }
+        return party.getPhoneAddresses().stream()
```

Comment:
Don't want the same "fallbackToFirst" behaviour?

Reply:
we don't have a fallback in this case because identity expects only a mobile number, with the ability to receive SMS codes.

Reply:
KPN used to have this service when an SMS was sent to a landline, the phone would ring and the message was read to you. I cannot verify it still works because I don't have a landline anymore. When asked, I will put my mobile number as "home" phone number. Shaky grounds.

Reply:
good point, I'll address that